packages/nc-gui/pages/[projectType]/form/[viewId]/index/index.vue

```diff
@@ -5,9 +5,32 @@ import { ref } from 'vue'
 import { StreamBarcodeReader } from 'vue-barcode-reader'
 import { useSharedFormStoreOrThrow } from '#imports'
 
-const { sharedFormView, submitForm, v$, formState, notFound, formColumns, submitted, secondsRemain, isLoading } =
+const { sharedFormView, submitForm, v$, formState, notFound, formColumns, submitted, secondsRemain, isLoading, sharedViewMeta } =
   useSharedFormStoreOrThrow()
 
+const route = useRoute()
+
+if (formColumns.value && sharedViewMeta.value.preFilledMode !== 'none' && Object.keys(route.query).length > 0) {
+  for (const column of formColumns.value) {
+    if (column.title && route.query[column.title]) {
+      if (isVirtualCol(column)) {
+        const virtData = (route.query[column.title] as string).split(';')
+        if (virtData.length) {
+          formState.value[column.title] = []
+          virtData.forEach((virtItem) => {
+            const virtRow = virtItem.split('|')
+            formState.value[column.title] = [...formState.value[column.title], { Id: virtRow[0], Title: virtRow[1] }]
```

Comment:
The `Id` nd `Title` should be replaced with the primary column and display column title of the related table. You can extract these values from the meta.

Reply:
OMG you're right. Not 100% familiar with the project yet; forgive my noob mistakes. Will finalize this PR during the weekend. Peace.

Reply:
And for belongs to it should be an object